From 0e6370ba644f9e7091cd1e2591b31ae40b2fc332 Mon Sep 17 00:00:00 2001
From: Mike Snitzer <snitzer@redhat.com>
Date: Fri, 13 Jun 2014 14:36:32 -0400
Subject: [block] blk-mq: handle NULL req return from blk_map_request in single queue mode

Message-id: <1402670238-13916-105-git-send-email-snitzer@redhat.com>
Patchwork-id: 83827
O-Subject: [RHEL7.1 PATCH 104/150] blk-mq: handle NULL req return from blk_map_request in single queue mode
Bugzilla: 1105204
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Jeff Moyer <jmoyer@redhat.com>

BZ: 1105204

Upstream commit ff87bcec197774f938fbd1fe996068005f3dfb3c
Author: Jens Axboe <axboe@fb.com>
Date:   Tue Jun 3 11:59:49 2014 -0600

    blk-mq: handle NULL req return from blk_map_request in single queue mode

    blk_mq_map_request() can return NULL if we fail entering the queue
    (dying, or removed), in which case it has already ended IO on the
    bio. So nothing more to do, except just return.

    Signed-off-by: Jens Axboe <axboe@fb.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/block/blk-mq.c b/block/blk-mq.c
index 3bb4cfe..96e6eb6 100644
--- a/block/blk-mq.c
+++ b/block/blk-mq.c
@@ -1276,6 +1276,8 @@ static void blk_sq_make_request(struct request_queue *q, struct bio *bio)
   return;
 
  rq = blk_mq_map_request(q, bio, &data);
+ if (unlikely(!rq))
+  return;
 
  if (unlikely(is_flush_fua)) {
   blk_mq_bio_to_request(rq, bio);
-- 
1.7.1